### ISSUE TITLE
Execute Amplify CLI locally

### ISSUE THREAD

**Issue by screendriver** [aws-amplify/amplify-cli#4491]
**Is your feature request related to a problem? Please describe.**

At the moment the Amplify CLI needs to be installed globally with `npm i -g @aws-amplify/cli`. It is not possible to use the CLI locally within the project. It gives weird errors or displays empty selections. Furthermore it is not possible to install the CLI globally via `Yarn`: `yarn global add @aws-amplify/cli`.

See also
https://github.com/aws-amplify/amplify-cli/issues/4484
https://github.com/aws-amplify/amplify-cli/issues/4485
https://github.com/aws-amplify/amplify-cli/issues/839

**Describe the solution you'd like**

The CLI should be able to be executed locally in my project so I can pin the version that is used within the project:

```sh
> npm i -D @aws-amplify/cli
> npx amplify
```

**Describe alternatives you've considered**

-

**Additional context**

Installing and executing binaries globally in a shared CI / CD environment is a bad practice which leads to unreliable non-reproducible builds and the most famous sentence `But it works on my machine`. Just imagine what happens if project A uses version 1 of a binary and project B, which runs in parallel, needs version 2.

**Comment by SwaySway** [comment]
Currently the amplify cli does not support local install we do have a PR in the works.
PR: https://github.com/aws-amplify/amplify-cli/pull/3857

Closing as a duplicate of #4043
